Source/NSOrderedSet.m

```diff
@@ -0,0 +1,1763 @@
+      [tmpArray addObject: obj];
+    }
+
+  // Move the objects
+  e = [tmpArray objectEnumerator];
+  while((o = [e nextObject]) != nil)
+    {
+      [self insertObject: o atIndex: index];
+    }
+
+  // Remove the originals...  
+  for(i = 0; i < count; i++)
+    {
+      NSUInteger index = indexArray[i];
+      [self removeObjectAtIndex: index];
+    }
+}
+
+- (void) exchangeObjectAtIndex:(NSUInteger)index withObjectAtIndex:(NSUInteger)otherIndex
+{
+  id	tmp = [self objectAtIndex: index];
+
+  RETAIN(tmp);
+  [self replaceObjectAtIndex: index withObject: [self objectAtIndex: otherIndex]];
+  [self replaceObjectAtIndex: otherIndex withObject: tmp];
+  RELEASE(tmp);
+}
+
+- (void)filterUsingPredicate:(NSPredicate *)predicate
+{
+  unsigned	count = [self count];
+
+  while (count-- > 0)
+    {
+      id	object = [self objectAtIndex: count];
+	
+      if ([predicate evaluateWithObject: object] == NO)
+        {
+          [self removeObjectAtIndex: count];
+        }
+    }
+}
+
+- (void) sortUsingDescriptors:(NSArray *)descriptors
+{
+  NSArray *result = [[self array] sortedArrayUsingDescriptors: descriptors];
+  [self removeAllObjects];
+  [self addObjectsFromArray: result];
+}
+
+- (void) sortUsingComparator: (NSComparator)comparator
+{
+  [self sortWithOptions: 0 usingComparator: comparator];
+}
+
+- (void) sortWithOptions: (NSSortOptions)options
+         usingComparator: (NSComparator)comparator
+{
+  NSUInteger count = [self count];
+  
+  if ((1 < count) && (NULL != comparator))
+    {
+      NSArray *res = nil;
+      NSUInteger i, c = [self count];
+      IMP	get = [self methodForSelector: oaiSel];
+
+      GS_BEGINIDBUF(objects, count);
+      for (i = 0; i < c; i++)
+	{
+	  objects[i] = (*get)(self, oaiSel, i);
+	}
+      
+      if (options & NSSortStable)
+        {
+          if (options & NSSortConcurrent)
+            {
+              GSSortStableConcurrent(objects, NSMakeRange(0,count),
+                (id)comparator, GSComparisonTypeComparatorBlock, NULL);
+            }
+          else
+            {
+              GSSortStable(objects, NSMakeRange(0,count),
+                (id)comparator, GSComparisonTypeComparatorBlock, NULL);
+            }
+        }
+      else
+        {
+          if (options & NSSortConcurrent)
+            {
+              GSSortUnstableConcurrent(objects, NSMakeRange(0,count),
+                (id)comparator, GSComparisonTypeComparatorBlock, NULL);
+            }
+          else
+            {
+              GSSortUnstable(objects, NSMakeRange(0,count),
+                (id)comparator, GSComparisonTypeComparatorBlock, NULL);
+            }
+        }
+      res = [[NSArray alloc] initWithObjects: objects count: count];
+      [self removeAllObjects];
+      [self addObjectsFromArray: res];
+
+      RELEASE(res);
+      GS_ENDIDBUF();
+    } 
+}
+
+- (void) sortRange: (NSRange)range
+           options:(NSSortOptions)options
+   usingComparator: (NSComparator)comparator
+{
+}
```

Comment:
The implementation is missing.

Reply:
Sorry, I don't get the concept of closing a conversation, just because you don't want to address it at the moment. The comments are the way to track outstanding issues.